### PR TITLE
[flutter_tools] Fix Android Studio minor patches detection on Windows

### LINKED ISSUE #70387: flutter doctor detecting wrong version of android Studio with 4.1.1 release
## Steps to Reproduce

1. Install Android Studio 4.1.1 and plugins
2. run `flutter doctor`
3. See that `Android Studio 4.1.0` is detected and plugins are not found.

## Logs

image

```
[√] Flutter (Channel stable, 1.22.3, on Microsoft Windows [Version 10.0.18363.1198], locale en-US)
    • Flutter version 1.22.3 at c:\flutter
    • Framework revision 8874f21e79 (2 weeks ago), 2020-10-29 14:14:35 -0700
    • Engine revision a1440ca392
    • Dart version 2.10.3

[√] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at C:\Users\jkolb\AppData\Local\Android\sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[!] Android Studio (version 4.1.0)
    • Android Studio at C:\Program Files\Android\Android Studio
    X Flutter plugin not installed; this adds Flutter specific functionality.
    X Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b01)

[√] VS Code (version 1.51.1)
    • VS Code at C:\Users\jkolb\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.16.0
```

### PR DESCRIPTION
**Pull request by TahaTesser**
https://github.com/flutter/flutter/issues/70387 was filed late last year, looking into it I found that It is not possible to fix this, you cannot correctly report minor patches in `flutter doctor -v` for that version. However, this seems to be possible now with the new Android Studio  4.2 release which generates this file -> `AppData\Local\Google\AndroidStudio4.2\.appinfo`, this file contains two tags that we can use to determine the Android Studio version correctly including minor patches.

`app.version.major=4&app.version.minor=2`

(Note: Currently this `\AndroidStudio4.2\.appinfo`  file still reports `4.2`, even tho the latest version is `4.2.1` but this PR would improve this version detection when the Android Studio team adds more minor patches info to the file `AppData\Local\Google\AndroidStudio4.2\.appinfo` file. Intellij's `.appinfo` file  already correctly reports minor version `2021.1.1` )

This PR will eliminate the need to increment minor patches detection for `Android Studio 4.2` releases

To test this behavior, I've manually edited `.appinfo` file, below is the result (for test purposes only)

### Before
```console
[✓] Android Studio (version 4.2.0)
    • Android Studio at C:\Users\Taha\Code\android-studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6842174)
```
### After 
```console
[√] Android Studio (version 4.2.1)
    • Android Studio at C:\Users\Taha\Code\android-studio
    • Flutter plugin can be installed from:
       https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
       https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6842174)
```

We should close https://github.com/flutter/flutter/issues/70387 as there is no way to fix this in an older version of Android Studio. ex: `Android Studio 4.1.1`

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter_tools/lib/src/android/android_studio.dart [hunk 1]
```diff
@@ -405,13 +408,25 @@ class AndroidStudio implements Comparable<AndroidStudio> {
...
         '.home',
       ));
-      if (homeDot.existsSync()) {
+      final File appInfo = globals.fs.file(globals.fs.path.join(
+        globals.platform.environment['LOCALAPPDATA']!,
+        'Google',
+        'AndroidStudio4.2',
+        '.appinfo',
+      ));
+      if (homeDot.existsSync() && appInfo.existsSync()) {
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/android/android_studio.dart:417]: Are we certain that the appinfo dot file exists for all versions that the home dot file exists for? i.e. did the first 4.2 have one?

Comment (TahaTesser) [reply packages/flutter_tools/lib/src/android/android_studio.dart:417]: I have tried 4.2 Canary release (which has `AndroidStudioPreview4.2` folder in LOCALAPPDATA) it doesn't appinfo dot file, but Android Studio 4.2 Beta 1  (which has `AndroidStudioPreview4.2` folder in LOCALAPPDATA as well) does have this but here in the source, we don't search for `AndroidStudioPreview4.2` folder, we only scan `AndroidStudio4.2` folder (stable Android Studio), which has this dot file too

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/android/android_studio.dart:417]: Then won't this fail to detect 4.2, since it needs both homedot and appinfo now?

Comment (TahaTesser) [reply packages/flutter_tools/lib/src/android/android_studio.dart:417]: yes

**Comment by TahaTesser** [comment]
@jonahwilliams 
Rethinking this, I feel it is better to add something more concrete where we can scan all Android Studio versions starting from 4.2 and add logic to scan newer versions such as 4.3, 4.4. so on automatically along with minor patches without scanning new release with more code 
I will work on this change in a new PR, closing this for now.

**Pull request closed without merging**